src/Control/Monad/Linear.hs

```diff
@@ -79,3 +80,13 @@ join action = action >>= id
 -- | Convenience operator to define Applicative instances in terms of Monad
 ap :: Monad m => m (a ->. b) ->. m a ->. m b
 ap f x = f >>= (\f' -> fmap f' x)
+
+-- | DerivingVia combinators for Data.XXX in terms of Control.XXX
+newtype DataFromControl f a = DataFromControl (f a)
```

Comment:
It's a bit wordy as it's supposed to be imported qualified. So how about `Control.Data` instead?

Reply:
Ah good point!